.github/workflows/sdist.yml

```diff
@@ -0,0 +1,51 @@
+name: SDist
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04,]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0 # needed for history/tags
+
+      # Used to host cibuildwheel
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12' 
+
+      - name: Get python_ics version
+        uses: mtkennerly/dunamai-action@v1
+        with:
+          env-var: PYTHON_ICS_VERSION
+          args: --format "v{base}-{commit}-{timestamp}"
+      - name: Print python_ics version
+        run: echo $PYTHON_ICS_VERSION
+      
+      - name: Install requirements for sdist building
+        run: python -m pip install -r requirements.txt
+
+      - name: Compiling sdist
+        run: python setup.py sdist
```

Comment:
This should probably use `python -m build --sdist`
https://build.pypa.io/en/stable/index.html

Reply:
Thanks, I'll look into it. I did build without and it gave me an error. I'm struggling right now trying to emulate arm builds since github doesn't provide linux arm runners.

Reply:
yeah since I depend on dunamai and it requires git it won't work right now. I'll have to figure that out first since it looks like `build` isolates the build environment.

Reply:
You can build for Linux ARM using cibildwheel and QEmu, see https://cibuildwheel.pypa.io/en/stable/faq/#emulation

For build, you can use the '--no-isolation' switch until you figure out the issue.

Reply:
@pierreluctg I managed to get the CI working for the arm builds. I tested this against our neoVI PI with the CM4 to verify it actually works. `--no-isolation` gave me the same error so I'll have to see what the deal is there later. I'll open an issue to fix it later.

Reply:
![Screenshot from 2024-05-24 23-52-36](https://github.com/intrepidcs/python_ics/assets/26259753/a4107f3b-fac1-4ba5-b0f3-60839779b529)

Reply:
https://github.com/intrepidcs/python_ics/issues/183

Reply:
Maybe consider moving from `dunamai` to `setuptools_scm` the PyPa recommended tool (https://setuptools-scm.readthedocs.io/en/stable/) ?